src/main/java/by/dragonsurvivalteam/dragonsurvival/common/entity/projectiles/BallLightningEntity.java

```diff
@@ -68,22 +68,35 @@ protected void onHit(HitResult p_70227_1_){
 				}
 			}
 			level.explode(null, damagesource, null, getX(), getY(), getZ(), explosivePower, flag, flag ? Explosion.BlockInteraction.DESTROY : Explosion.BlockInteraction.NONE);
+			if (!(getOwner() instanceof Player))
+			{
+				level.playLocalSound(getX(), getY(), getZ(), SoundEvents.LIGHTNING_BOLT_IMPACT, SoundSource.HOSTILE, 3.0F, 0.5f, false);
+				return;
+			}
 
 			isDead = true;
 			setDeltaMovement(0, 0, 0);
 
 			attackMobs();
 		}
 	}
+	
+	@Override
+	public void tick() {
+		super.tick();
+		if (level.dayTime() % 5 == 0 && !isDead) // Once per 5 ticks (0.25 seconds)
+			attackMobs();
+	}
 
 	@Override
 	public void attackMobs(){
-		if (!(getOwner() instanceof Player))
-		{
-			level.playLocalSound(getX(), getY(), getZ(), SoundEvents.LIGHTNING_BOLT_IMPACT, SoundSource.HOSTILE, 3.0F, 0.5f, false);
-			return;
+		int rn;
+		try {
+			rn = DragonAbilities.getSelfAbility((Player)getOwner(), BallLightningAbility.class).getRange();
+		} catch (java.lang.ClassCastException error) {
+			rn = 4;
```

Comment:
recommend doing sth. like this instead of running into exceptions

```java
int rn = 4;
whatever_type_this_is = getOwner();

if (whatever_type_this_is instanceof Player player) {
    rn = DragonAbilities.getSelfAbility(player, BallLightningAbility.class).getRange();
}
```

Reply:
Thank you.  I've updated the file accordingly.